src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const changesArray = [];
+  let currentState = { ...state };
```

Comment:
The `currentState` is correctly initialized as a clone of the initial state to ensure that the original state is not mutated.